include/internal/ktls.h

```diff
@@ -67,11 +68,20 @@ static ossl_inline int ktls_send_ctrl_message(int fd, unsigned char record_type,
     return -1;
 }
 
+
+static ossl_inline int ktls_read_record(int fd, void *data, size_t length)
+{
+    return -1;
+}
+
 #   else                        /* KERNEL_VERSION */
 
 #    include <netinet/tcp.h>
 #    include <linux/tls.h>
 #    include <linux/socket.h>
+#    include "openssl/ssl3.h"
+#    include "openssl/tls1.h"
+#    include "openssl/evp.h"
 
 #    ifndef SOL_TLS
 #     define SOL_TLS 282
```

Comment:
Do you see a chance that at least the TCP_ULP below can be used from the linux headers?

Reply:
I've tried doing this, but it creates conflicts with libc includes. As a result, I try to avoid including the kernel header that holds this. I think that the solution for this one is to wait until libc fetches the necessary headers from the kernel.

Reply:
OK. Thanks for trying.